// roombridge-client/src/pages/profile.js
import React, { useContext, useState, useEffect } from "react";
import toast from "react-hot-toast";
import { UserContext } from "../context/userContext";

const ProfilePage = () => {
  const { user, setUser } = useContext(UserContext);
  const [isEditing, setIsEditing] = useState(false);
  const [userStats, setUserStats] = useState({
    roomsCount: 0,
    rating: 0
  });
  const [formData, setFormData] = useState({
    full_name: user?.full_name || "",
    username: user?.username || "",
    email: user?.email || "",
    bio: user?.bio || "",
    mobile_number: user?.mobile_number || "",
    gender: user?.gender || ""
  });

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setFormData(prev => ({
      ...prev,
      [name]: value
    }));
  };

  // Fetch user statistics
  const fetchUserStats = async () => {
    try {
      const token = localStorage.getItem('token');
      const response = await fetch('http://localhost:8000/api/v1/rooms/user-rooms', {
        method: 'GET',
        headers: {
          'Authorization': `Bearer ${token}`
        }
      });

      if (response.ok) {
        const result = await response.json();
        setUserStats({
          roomsCount: result.data?.length || 0,
          rating: 4.5 // You can calculate this from reviews later
        });
      }
    } catch (error) {
      console.error("Error fetching user stats:", error);
    }
  };

  // Load user stats when component mounts
  useEffect(() => {
    if (user) {
      fetchUserStats();
    }
  }, [user]);

  const handleSaveProfile = async () => {
    try {
      const token = localStorage.getItem('token');
      const response = await fetch('http://localhost:8000/api/v1/user/profile', {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        },
        body: JSON.stringify(formData)
      });

      if (response.ok) {
        const result = await response.json();
        setUser(result.data);
        setIsEditing(false);
        toast.success("Profile updated successfully!");
      } else {
        toast.error("Failed to update profile");
      }
    } catch (error) {
      console.error("Error updating profile:", error);
      toast.error("Error updating profile");
    }
  };

  const handleCopyLink = () => {
    if (user && user._id) {
      const referralLink = `https://roombridge.vercel.app/register-user?referralLink=${user._id}`;
      navigator.clipboard
        .writeText(referralLink)
        .then(() => {
          toast.success("Referral link copied to clipboard");
        })
        .catch((err) => {
          console.error("Failed to copy the link: ", err);
        });
    } else {
      toast.error("Failed to generate referral link");
    }
  };
  return (
    <>
      <div className="flex-col md:w-1/2 sm:w-1/2 w-full items-center flex sm:mt-4 mt-0">
        <main className=" sm:w-1/2 w-full max-h-screen">
          <div className="px-6 sm:py-8 ">
            <div className=" mx-auto">
              <div className="bg-white rounded-3xl p-8 mb-5">
                <div className="flex justify-between items-start mb-4">
                  <div>
                    <img
                      className="w-20 h-20 rounded-full"
                      src={user?.profile_picture_url || "https://avatars.githubusercontent.com/u/82640789?v=4"}
                      alt="Profile avatar"
                    />
                  </div>
                  <button
                    onClick={() => setIsEditing(!isEditing)}
                    className="px-4 py-2 bg-blue-500 text-white rounded-lg hover:bg-blue-600"
                  >
                    {isEditing ? "Cancel" : "Edit Profile"}
                  </button>
                </div>

                {isEditing ? (
                  <div className="space-y-4">
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Full Name</label>
                      <input
                        type="text"
                        name="full_name"
                        value={formData.full_name}
                        onChange={handleInputChange}
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      />
                    </div>
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Username</label>
                      <input
                        type="text"
                        name="username"
                        value={formData.username}
                        onChange={handleInputChange}
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      />
                    </div>
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Email</label>
                      <input
                        type="email"
                        name="email"
                        value={formData.email}
                        onChange={handleInputChange}
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      />
                    </div>
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Bio</label>
                      <textarea
                        name="bio"
                        value={formData.bio}
                        onChange={handleInputChange}
                        rows="3"
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      />
                    </div>
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Mobile Number</label>
                      <input
                        type="text"
                        name="mobile_number"
                        value={formData.mobile_number}
                        onChange={handleInputChange}
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      />
                    </div>
                    <div>
                      <label className="block text-sm font-medium text-gray-700">Gender</label>
                      <select
                        name="gender"
                        value={formData.gender}
                        onChange={handleInputChange}
                        className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
                      >
                        <option value="">Select Gender</option>
                        <option value="male">Male</option>
                        <option value="female">Female</option>
                      </select>
                    </div>
                    <button
                      onClick={handleSaveProfile}
                      className="w-full px-4 py-2 bg-green-500 text-white rounded-lg hover:bg-green-600"
                    >
                      Save Changes
                    </button>
                  </div>
                ) : (
                  <div>
                    <h1 className="text-xl font-bold mt-4">{user?.full_name || "User Name"}</h1>
                    <p className="text-gray-600 mt-1">@{user?.username || "username"}</p>
                    <p className="text-gray-500 text-sm">{user?.email || "email@example.com"}</p>
                    <p className="mt-2">
                      {user?.bio || "✅ Full Stack Developer ✅ Frontend Development ✅ Backend Development ✅ Blockchain Developer ✅ Integration Web2 to Web3 / Wallet Integration ✅"}
                    </p>
                  </div>
                )}

                <div className="flex gap-3 mt-4">
                  <div className="h-6 w-6">
                    <img
                      className="object-cover w-full h-full rounded-full"
                      src="https://ui-avatars.com/api/?background=random"
                    />
                  </div>
                  <div className="h-6 w-6">
                    <img
                      className="object-cover w-full h-full rounded-full"
                      src="https://ui-avatars.com/api/?background=random"
                    />
                  </div>
                  <div className="h-6 w-6">
                    <img
                      className="object-cover w-full h-full rounded-full"
                      src="https://ui-avatars.com/api/?background=random"
                    />
                  </div>
                </div>
                <div className="flex items-center justify-between mt-10">
                  <div className="flex items-stretch">
                    <div className="text-gray-400 text-xs">
                      Members
                      <br />
                      connected
                    </div>
                    <div className="h-100 border-l mx-4"></div>
                    <div className="flex flex-nowrap -space-x-3">
                      <div className="h-9 w-9">
                        <img
                          className="object-cover w-full h-full rounded-full"
                          src="https://ui-avatars.com/api/?background=random"
                        />
                      </div>
                      <div className="h-9 w-9">
                        <img
                          className="object-cover w-full h-full rounded-full"
                          src="https://ui-avatars.com/api/?background=random"
                        />
                      </div>
                    </div>
                  </div>
                  <div className="flex items-center gap-x-2">
                    <button
                      type="button"
                      className="inline-flex items-center justify-center h-9 px-3 rounded-xl border hover:border-gray-400 text-gray-800 hover:text-gray-900 transition"
                    >
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        width="1em"
                        height="1em"
                        fill="currentColor"
                        className="bi bi-chat-fill"
                        viewBox="0 0 16 16"
                      >
                        <path d="M8 15c4.418 0 8-3.134 8-7s-3.582-7-8-7-8 3.134-8 7c0 1.76.743 3.37 1.97 4.6-.097 1.016-.417 2.13-.771 2.966-.079.186.074.394.273.362 2.256-.37 3.597-.938 4.18-1.234A9.06 9.06 0 0 0 8 15z" />
                      </svg>
                    </button>
                    <button
                      type="button"
                      className="inline-flex items-center justify-center h-9 px-5 rounded-xl bg-gray-900 text-gray-300 hover:text-white text-sm font-semibold transition"
                    >
                      Open
                    </button>
                  </div>
                </div>

                <hr className="my-6" />

                <div className="grid sm:grid-cols-2 grid-cols-1 gap-x-20">
                  <div>
                    <h2 className="sm:text-2xl text-xl uppercase font-bold mb-4">Stats</h2>

                    <div className="grid grid-cols-2 gap-4">
                      <div className="col-span-2">
                        <div className="p-4 bg-green-100 rounded-xl">
                          <div className="font-bold text-xl text-gray-800 leading-none">
                            Good day, <br />
                            {user?.full_name || user?.username || "User"}
                          </div>
                          <div className="mt-5">
                            <button
                              type="button"
                              onClick={handleCopyLink}
                              className="inline-flex items-center justify-center py-2 px-3 rounded-xl bg-white text-gray-800 hover:text-green-500 text-sm font-semibold transition"
                            >
                              Referral Link
                            </button>
                          </div>
                        </div>
                      </div>
                      <div className="p-4 bg-yellow-100 rounded-xl text-gray-800">
                        <div className="font-bold text-2xl leading-none">
                          {userStats.roomsCount}
                        </div>
                        <div className="mt-2">Rooms Listed</div>
                      </div>
                      <div className="p-4 bg-yellow-100 rounded-xl text-gray-800">
                        <div className="font-bold text-2xl leading-none">
                          {userStats.rating || "N/A"}
                        </div>
                        <div className="mt-2">Ratings</div>
                      </div>
                      <div className="col-span-2">
                        <div className="p-4 bg-purple-100 rounded-xl text-gray-800">
                          <div className="font-bold text-xl leading-none">
                            Preferences
                          </div>
                         <div className="flex-wrap flex gap-y-3 mt-4"> 
                           {user?.preferences?.preferences?.map((preference, index) => (
                              <span key={index} className="bg-purple-100 text-purple-800 text-xs font-medium me-2 px-2.5 py-0.5 rounded dark:bg-gray-700 dark:text-purple-400 border border-purple-400">
                                {preference}
                              </span>
                           )) || <span className="text-gray-500 text-sm">No preferences set</span>}
                         </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <h2 className="sm:text-2xl text-xl uppercase font-bold mb-4 mt-8 sm:mt-0">
                      Your Rooms History
                    </h2>

                    <div className="space-y-4">
                      <div className="p-4 bg-white border rounded-xl text-gray-800 space-y-2">
                        <div className="flex justify-between">
                          <div className="text-gray-400 text-xs">Number 10</div>
                          <div className="text-gray-400 text-xs">4h</div>
                        </div>
                        <a
                          href="javascript:void(0)"
                          className="font-bold hover:text-yellow-800 hover:underline"
                        >
                          Blog and social posts
                        </a>
                        <div className="text-sm text-gray-600">
                          <svg
                            xmlns="http://www.w3.org/2000/svg"
                            width="1em"
                            height="1em"
                            fill="currentColor"
                            className="text-gray-800 inline align-middle mr-1"
                            viewBox="0 0 16 16"
                          >
                            <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8 4a.905.905 0 0 0-.9.995l.35 3.507a.552.552 0 0 0 1.1 0l.35-3.507A.905.905 0 0 0 8 4zm.002 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2z" />
                          </svg>
                          Deadline is today
                        </div>
                      </div>
                      <div className="p-4 bg-white border rounded-xl text-gray-800 space-y-2">
                        <div className="flex justify-between">
                          <div className="text-gray-400 text-xs">
                            Grace Aroma
                          </div>
                          <div className="text-gray-400 text-xs">7d</div>
                        </div>
                        <a
                          href="javascript:void(0)"
                          className="font-bold hover:text-yellow-800 hover:underline"
                        >
                          New campaign review
                        </a>
                        <div className="text-sm text-gray-600">
                          <svg
                            xmlns="http://www.w3.org/2000/svg"
                            width="1em"
                            height="1em"
                            fill="currentColor"
                            className="text-gray-800 inline align-middle mr-1"
                            viewBox="0 0 16 16"
                          >
                            <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8 4a.905.905 0 0 0-.9.995l.35 3.507a.552.552 0 0 0 1.1 0l.35-3.507A.905.905 0 0 0 8 4zm.002 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2z" />
                          </svg>
                          New feedback
                        </div>
                      </div>
                      <div className="p-4 bg-white border rounded-xl text-gray-800 space-y-2">
                        <div className="flex justify-between">
                          <div className="text-gray-400 text-xs">Petz App</div>
                          <div className="text-gray-400 text-xs">2h</div>
                        </div>
                        <a
                          href="javascript:void(0)"
                          className="font-bold hover:text-yellow-800 hover:underline"
                        >
                          Cross-platform and browser QA
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </>
  );
};

export default ProfilePage;
